homeassistant/components/otbr/config_flow.py

```diff
@@ -101,5 +123,142 @@ async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResu
+
+        try:
+            if otbr_dataset_tlv := await data.get_active_dataset_tlvs():
+                otbr_dataset = tlv_parser.parse_tlv(otbr_dataset_tlv.hex())
+        except HomeAssistantError:
+            _LOGGER.warning("Could not read active dataset", exc_info=True)
+
+        if preferred_dataset_tlv := await async_get_preferred_dataset(self.hass):
+            preferred_dataset = tlv_parser.parse_tlv(preferred_dataset_tlv)
+
+        if otbr_dataset and otbr_dataset != preferred_dataset:
+            menu_options.append("prefer_otbr_network")
+            placeholders["otbr_network"] = otbr_dataset.get(
+                tlv_parser.MeshcopTLVType.NETWORKNAME, ""
+            )
+
+        if preferred_dataset and preferred_dataset != otbr_dataset:
+            menu_options.append("use_preferred_network")
+            placeholders["preferred_network"] = preferred_dataset.get(
+                tlv_parser.MeshcopTLVType.NETWORKNAME, ""
+            )
+
+        self.context["placeholders"] = placeholders
+        return self.async_show_menu(
+            step_id="thread_network_menu",
+            menu_options=menu_options,
+            description_placeholders=placeholders,
+        )
+
+    async def async_step_create_network(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Create a new network, and optionally mark it as preferred."""
+        if user_input is not None:
+            data: OTBRData = self.hass.data[DOMAIN]
+
+            # We currently have no way to know which channel zha is using, assume it's
+            # the default
+            zha_channel = DEFAULT_CHANNEL
+
+            try:
+                # Disable the router, create a new network and read it back
+                await data.set_enabled(False)
+                await data.create_active_dataset(
+                    python_otbr_api.OperationalDataSet(
+                        channel=zha_channel, network_name="home-assistant"
+                    )
+                )
+                await data.set_enabled(True)
+                dataset_tlvs = await data.get_active_dataset_tlvs()
+            except HomeAssistantError:
+                _LOGGER.warning("Failed to create new Thread network", exc_info=True)
+                return self.async_abort(reason="unknown")
+            if not dataset_tlvs:
+                _LOGGER.warning("Got empty network")
+                return self.async_abort(reason="unknown")
+
+            dataset_id = await async_add_dataset(
+                self.hass, self._config_entry.title, dataset_tlvs.hex()
+            )
+            await async_set_preferred_dataset(self.hass, dataset_id)
+
+            return self.async_create_entry(data={})
+
+        return self.async_show_form(step_id="create_network")
+
+    async def async_step_prefer_otbr_network(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Make the OTBR's network the preferred network."""
```

Comment:
This is not the OTBR responsibility. That's for the Thread integration and panel to provide.